### PR TITLE
29 8 merchant item update

### LINKED ISSUE #29: 8 Merchant Item Update
As a merchant,
When I visit the merchant show page of an item (/merchants/:merchant_id/items/:item_id)
I see a link to update the item information.
When I click the link
Then I am taken to a page to edit this item
And I see a form filled in with the existing item attribute information
When I update the information in the form and I click ‘submit’
Then I am redirected back to the item show page where I see the updated information
And I see a flash message stating that the information has been successfully updated.

### PR DESCRIPTION
**Pull request by Jredish11**
This PR adds all tests, views, controller actions, validations, validation tests for the Merchant Items update.

### CODE REVIEW

File: app/controllers/merchants/items_controller.rb [hunk 1]
```diff
@@ -4,6 +4,29 @@ def index
   end
 
   def show
+    @merchant = Merchant.find(params[:merchant_id])
```
Comment (Jredish11) [inline app/controllers/merchants/items_controller.rb:7]: need to fix my show view and add item variable here.

File: app/views/merchants/items/show.html.erb [hunk 2]
```diff
@@ -1,4 +1,7 @@
 <h2>Item Attributes</h2>
-<p> Name:<%= @item.name %></p>
-<p>Description:<%= @item.description %></p>
-<p>Current Selling Price:<%= @item.unit_price %></p>
\ No newline at end of file
+  <% @merchant.items.each do |item| %>
```
Comment (Jredish11) [inline app/views/merchants/items/show.html.erb]: need to fix this to correct render issue.

File: app/controllers/merchants/items_controller.rb [hunk 3]
```diff
@@ -4,6 +4,30 @@ def index
...
+
+  def edit
+    @merchant = Merchant.find(params[:merchant_id])
+    @item = Item.find(params[:id])
+  end
+
+  def update
     @item = Item.find(params[:id])
+    @merchant = Merchant.find(params[:merchant_id])
+    if @item.update(item_params)
```
Comment (stackmm) [inline app/controllers/merchants/items_controller.rb:19]: nice, flash messages show up just as described in the user story

File: app/views/merchants/items/show.html.erb [hunk 4]
```diff
@@ -1,4 +1,6 @@
 <h2>Item Attributes</h2>
-<p> Name:<%= @item.name %></p>
-<p>Description:<%= @item.description %></p>
-<p>Current Selling Price:<%= @item.unit_price %></p>
\ No newline at end of file
+
+  <p> Name:<%= @item.name %></p>
+  <p>Description:<%= @item.description %></p>
+  <p>Current Selling Price:<%= @item.unit_price %></p>
+  <p><%= link_to "Update #{@item.name}", edit_merchant_item_path(@merchant, @item) %></p>
```
Comment (stackmm) [inline app/views/merchants/items/show.html.erb:6]: the wireframe shows the button can just be "Update Item" for every item (dont need to interpolate in the item's name on the button)

**Review by stackmm — APPROVED** [review]

**Pull request merged**